### PR TITLE
Add new deviation BackupNHGRequiresVrfWithDecap

### PR DESCRIPTION
**Pull request by Ankur19**
Some devices can't install Backup Next Hops pointing only to a VRF. 

For such scenarios these devices require the Next Hop to define VRF + Decap.

### CODE REVIEW

**Comment by Ankur19** [comment]
@prakashbadri-arista @bstoll Please take a look

**Review by bstoll — APPROVED** [review]

File: feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go [hunk 1]
```diff
@@ -127,6 +130,24 @@ func TestDirectBackupNexthopGroup(t *testing.T) {
...
 	configureDUT(t, dut)
 	configureNetworkInstance(t, dut)
+	// For interface configuration, prefer config Vrf first then the IP address
+	if deviations.InterfaceConfigVRFBeforeAddress(dut) {
+		configureDUT(t, dut)
+	}
+
+	if deviations.BackupNHGRequiresVrfWithDecap(dut) {
+		d := &oc.Root{}
+		ni := d.GetOrCreateNetworkInstance(vrfA)
```
Comment (prakashbadri-arista) [inline feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go:140]: Typically the vrf selection policy is only created in the DEFAULT NI.  did this work in EOS ?

File: feature/experimental/backup_nhg/otg_tests/backup_nhg_test/backup_nhg_test.go [hunk 2]
```diff
@@ -281,19 +307,24 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
...
+	nh2, op2 := gribi.NHEntry(nh2ID, atePort3.IPv4, deviations.DefaultNetworkInstance(a.dut), fluent.InstalledInFIB)
+	nhg1, op3 := gribi.NHGEntry(nhg1ID, map[uint64]uint64{nh1ID: 100}, deviations.DefaultNetworkInstance(a.dut), fluent.InstalledInFIB)
+	nhg2, op4 := gribi.NHGEntry(nhg2ID, map[uint64]uint64{nh2ID: 100}, deviations.DefaultNetworkInstance(a.dut), fluent.InstalledInFIB)
+	a.client.AddEntries(t, []fluent.GRIBIEntry{nh1, nh2, nhg1, nhg2}, []*client.OpResult{op1, op2, op3, op4})
 	t.Logf("Adding an IPv4Entry for %s via gRIBI", nhip)
 	a.client.AddIPv4(t, nhip+"/"+mask, nhg1ID, deviations.DefaultNetworkInstance(a.dut), deviations.DefaultNetworkInstance(a.dut), fluent.InstalledInFIB)
 	t.Logf("Adding NH %d in VRF-B via gRIBI", nh100ID)
-	a.client.AddNH(t, nh100ID, "VRFOnly", deviations.DefaultNetworkInstance(a.dut), fluent.InstalledInFIB, &gribi.NHOptions{VrfName: vrfB})
+	nh100, op5 := gribi.NHEntry(nh100ID, "VRFOnly", deviations.DefaultNetworkInstance(a.dut), fluent.InstalledInFIB, &gribi.NHOptions{VrfName: vrfB})
+	if deviations.BackupNHGRequiresVrfWithDecap(a.dut) {
```
Comment (prakashbadri-arista) [inline feature/experimental/backup_nhg/otg_tests/backup_nhg_test/backup_nhg_test.go:320]: I would think that this is a necessary validation for all vendors. Any reason why went to exercise this usecase only with a deviation ?

File: feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go (hunk 1 above)
Comment (Ankur19) [reply feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go:140]: It is working. Is there a way to do this using Default VRF without changing the Test plan?

File: feature/experimental/backup_nhg/otg_tests/backup_nhg_test/backup_nhg_test.go (hunk 2 above)
Comment (Ankur19) [reply feature/experimental/backup_nhg/otg_tests/backup_nhg_test/backup_nhg_test.go:320]: For other vendors it is still `NH --> VRF`, for Arista we are exercising `NH --> {VRF + Decap}`

Comment (Ankur19) [reply feature/experimental/backup_nhg/otg_tests/backup_nhg_test/backup_nhg_test.go:320]: We don't want to change the test case.

**Review by bstoll — APPROVED** [review]

**Pull request merged**